Models/IonAuthModel.php

```diff
@@ -374,13 +374,13 @@ public function getUserByActivationCode(string $userCode)
 	 * Validates and removes activation code.
 	 *
 	 * @param integer|string $id   The user identifier
-	 * @param boolean        $code The *user* activation code
+	 * @param string         $code The *user* activation code
```

Comment:
I Think `@param boolean|string $code ...` will be better

Reply:
I don't think so, because the function should only accept a string or nothing and nothing isn't a boolean.

Reply:
OK, so in this case, the function must be
`public function activate($id, ?string $code=null): bool`

And the follwing code is false : `$code === false`

Reply:
Was the ? a mistake?

Reply:
No : [https://www.php.net/manual/en/migration71.new-features.php](https://www.php.net/manual/en/migration71.new-features.php)

Reply:
Thanks for the information.
I committed a new version that contains your suggestion.